service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -0,0 +1,781 @@
+  curl --retry 5 -s -f \
+    -H "Metadata-Flavor: Google" \
+    "http://metadata/computeMetadata/v1/${metadata_path}"
+}
+
+#######################################
+# Set guest attributes on GCE. Used here to log completion status of the script.
+# See https://cloud.google.com/compute/docs/metadata/manage-guest-attributes
+# Arguments:
+#   $1: The guest attribute domain and key IE startup_script/status
+#   $2  The data to write to the guest attribute
+#######################################
+function set_guest_attributes() {
+  local attr_path="${1}"
+  local attr_value="${2}"
+  curl -s -X PUT --data "${attr_value}" \
+    -H "Metadata-Flavor: Google" \
+    "http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/${attr_path}"
+}
+
+# If the script exits without error let the UI know it completed successfully
+# Otherwise if an error occurred write the line and command that failed to guest attributes.
+function exit_handler {
+  local exit_code="${1}"
+  local line_no="${2}"
+  local command="${3}"
+  # Success! Set the guest attributes and exit cleanly
+  if [[ "${exit_code}" -eq 0 ]]; then
+    set_guest_attributes "${STATUS_ATTRIBUTE}" "COMPLETE"
+    exit 0
+  fi
+  # Write error status and message to guest attributes
+  set_guest_attributes "${STATUS_ATTRIBUTE}" "ERROR"
+  set_guest_attributes "${MESSAGE_ATTRIBUTE}" "Error on line ${line_no}, command \"${command}\". See ${POST_STARTUP_OUTPUT_FILE} for more information."
+  exit "${exit_code}"
+}
+trap 'exit_handler $? $LINENO $BASH_COMMAND' EXIT
+
+#######################################
+### Begin environment setup 
+#######################################
+
+# Let the UI know the script has started
+set_guest_attributes "${STATUS_ATTRIBUTE}" "STARTED"
+
+# Add dataproc user to sudoers
+emit "Adding dataproc user to sudoers"
+sudo usermod -aG sudo $DATAPROC_USER
+
+# Remove default user bashrc to ensure that the user's bashrc is sourced in non interactive shells
+${RUN_AS_DATAPROC_USER} "rm -f '${USER_BASHRC}'"
+${RUN_AS_DATAPROC_USER} "touch '${USER_BASHRC}'"
+
+emit "Resynchronizing apt package index..."
+
+# The apt package index may not be clean when we run; resynchronize
+apt-get update
+
+# Create the target directories for installing into the HOME directory
+${RUN_AS_DATAPROC_USER} "mkdir -p '${USER_BASH_COMPLETION_DIR}'"
+${RUN_AS_DATAPROC_USER} "mkdir -p '${USER_HOME_LOCAL_BIN}'"
+${RUN_AS_DATAPROC_USER} "mkdir -p '${USER_HOME_LOCAL_SHARE}'"
+
+# As described above, have the ~/.bash_profile source the ~/.bashrc
+cat << EOF >> "${USER_BASH_PROFILE}"
+
+### BEGIN: Terra-specific customizations ###
+if [[ -e ~/.bashrc ]]; then
+  source ~/.bashrc
+fi
+### END: Terra-specific customizations ###
+
+EOF
+
+# Indicate the start of Terra customizations of the ~/.bashrc
+cat << EOF >> "${USER_BASHRC}"
+
+### BEGIN: Terra-specific customizations ###
+
+# Set the correct java installation to use
+export JAVA_HOME="${USER_HOME_LOCAL_DIR}"
```

Comment:
Also had trouble getting the dataproc user pointing to the correct java binary instead of the default system java binary..